casper/src/test/scala/coop/rchain/casper/MultiParentCasperMergeSpec.scala

```diff
@@ -58,11 +61,19 @@ class MultiParentCasperMergeSpec extends FlatSpec with Matchers with Inspectors
 
   it should "handle multi-parent blocks correctly when they operate on stdout" ignore effectTest {
     def echoContract(no: Int) = s"""new stdout(`rho:io:stdout`) in { stdout!("Contract $no") }"""
-    val time                  = System.currentTimeMillis()
+    testConflictOnSimpleDiamondDag(echoContract)
+  }
+
+  it should "handle multi-parent blocks correctly when they operate on volatile produce/consume pairs" in effectTest {
+    testConflictOnSimpleDiamondDag((no: Int) => s"""@"hi"!($no) | for (_ <- @"hi") { Nil }""")
+  }
+
+  private def testConflictOnSimpleDiamondDag(contract: Int => String): Effect[Unit] = {
+    val time = System.currentTimeMillis()
     HashSetCasperTestNode.networkEff(genesis, networkSize = 2).use { nodes =>
       val deploys = Vector(
-        ConstructDeploy.sourceDeploy(echoContract(1), time + 1, accounting.MAX_VALUE),
-        ConstructDeploy.sourceDeploy(echoContract(2), time + 2, accounting.MAX_VALUE)
+        ConstructDeploy.sourceDeploy(contract(1), time + 1, accounting.MAX_VALUE),
+        ConstructDeploy.sourceDeploy(contract(2), time + 2, accounting.MAX_VALUE)
```

Comment:
We've got way too few tests for conflict detection, and they should not be made in this class - at least not primarily. Each such test takes 1.5 s here, could take 15ms in the other class.

Please introduce the test methods I've menioned above in the EstimatorHelperTest, and go through each case of first 5 column's of Mike Stay's spreadsheet I've linked above - this will amount to ~30 cases:

15 cells from the spreadsheet * 2 for whether the 'existing' side of the match is in BASE or in one of the children (let's always make it A in the test data and make the test methods test twice by swapping the test data).

Reply:
I think the merge spec is the place these tests should be as long as they are using networkEff.\